fearless/ApplicationLayer/Services/WalletBalanceFetcher.swift

```diff
@@ -0,0 +1,405 @@
+import Foundation
+import RobinHood
+
+// swiftlint:disable file_length
+typealias WalletBalanceResult = Result<[String: WalletBalance], Error>
+typealias WalletBalanceResultCompletion = (WalletBalanceResult) -> Void
+
+protocol WalletBalanceFetcherProtocol {
+    /// Collects and counts all the information for `WalletBalance`
+    /// - Parameters:
+    ///   - walletId: Balance for specific wallet Id. If nil calculating for all meta accounts
+    ///   - completion: Called when WalletBalance will calculated
+    ///   - queue: The queue to which the result will be delivered
+    func fetchBalanceFor(
```

Comment:
please rename to 'subsctibeWalletBalance' and use delegate instead of completion.
also please handle events like 'selected account changed', 'enabled assets changed for wallet'